hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodesInPath.java

```diff
@@ -540,6 +540,20 @@ private String toString(boolean vaildateObject) {
     return b.toString();
   }
 
+  void returnToPool() {
+    // return objects back to pool
+    for (int i = 0; i < inodes.length - 1; ++i) {
```

Comment:
returnToPool or reduce atomic counts

Reply:
page: 33

https://www.slideshare.net/sjlee0/robust-and-scalable-concurrent-programming-lesson-from-the-trenches